File: MetodosOrdenamiento/hash/hash.py
import random

def hash_sort(arr):
    if len(arr) == 0:
        return arr
    
    min_val = min(arr)
    max_val = max(arr)
    
    # Crear tabla hash
    hash_table = {}
    
    # Insertar elementos en la tabla hash
    for num in arr:
        if num in hash_table:
            hash_table[num] += 1
        else:
            hash_table[num] = 1
    
    # Reconstruir array ordenado
    sorted_arr = []
    for num in range(min_val, max_val + 1):
        if num in hash_table:
            sorted_arr.extend([num] * hash_table[num])
    
    return sorted_arr

# Generar arreglo de 10 elementos aleatorios
arr = [random.randint(1, 100) for _ in range(10)]
print("Array original:", arr)

# Ordenar con Hash Sort
sorted_arr = hash_sort(arr)
print("Array ordenado:", sorted_arr)

File: MetodosOrdenamiento/hash/hash.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>
#include <random>

using namespace std;

vector<int> hash_sort(const vector<int>& arr) {
    if (arr.empty()) return arr;
    
    int min_val = *min_element(arr.begin(), arr.end());
    int max_val = *max_element(arr.begin(), arr.end());

    // Crear tabla hash
    unordered_map<int, int> hash_table;
    
    // Insertar elementos en la tabla hash
    for (int num : arr) {
        hash_table[num]++;
    }
    
    // Reconstruir vector ordenado
    vector<int> sorted_arr;
    for (int num = min_val; num <= max_val; ++num) {
        if (hash_table.find(num) != hash_table.end()) {
            for (int i = 0; i < hash_table[num]; ++i) {
                sorted_arr.push_back(num);
            }
        }
    }
    
    return sorted_arr;
}

int main() {
    // Generar arreglo de 10 elementos aleatorios
    vector<int> arr(10);
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> dis(1, 100);
    
    for (int i = 0; i < 10; ++i) {
        arr[i] = dis(gen);
    }
    
    cout << "Array original: ";
    for (int num : arr) {
        cout << num << " ";
    }
    cout << endl;
    
    // Ordenar con Hash Sort
    vector<int> sorted_arr = hash_sort(arr);
    
    cout << "Array ordenado: ";
    for (int num : sorted_arr) {
        cout << num << " ";
    }
    cout << std::endl;
    
    return 0;
}